hazelcast-it/distribution-it/src/test/java/com/hazelcast/it/hibernate/HibernateIT.java

```diff
@@ -0,0 +1,130 @@
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.NightlyTest;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.cfg.Environment;
+import org.hibernate.stat.Statistics;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(NightlyTest.class)
+public class HibernateIT extends HazelcastTestSupport {
+
+    private static final ILogger log = Logger.getLogger(HibernateIT.class);
+
+    private Process process;
+    private String clusterName;
+    private SessionFactory sessionFactory;
+
+    @Before
+    public void setUp() throws Exception {
+        assumeThatNoWindowsOS();
+        clusterName = this.getClass().getSimpleName() + "_" + UUID.randomUUID();
+        startHzMember();
+        sessionFactory = createSessionFactory(getCacheProperties());
+    }
+
+    @Test
+    public void testInsertLoad() {
+        Session session = sessionFactory.openSession();
+        Transaction tx = session.beginTransaction();
+        AnnotatedEntity e = new AnnotatedEntity("some-title");
+        session.save(e);
+        tx.commit();
+        session.close();
+
+        for (int i = 0; i < 10; i++) {
+            session = sessionFactory.openSession();
+            AnnotatedEntity retrieved = session.get(AnnotatedEntity.class, (long) 1);
+            assertThat(retrieved.getTitle()).isEqualTo("some-title");
+            session.close();
+        }
+
+        Statistics stats = sessionFactory.getStatistics();
+        assertThat(stats.getEntityInsertCount()).isEqualTo(1);
+        assertThat(stats.getSecondLevelCachePutCount()).isEqualTo(1);
+        assertThat(stats.getSecondLevelCacheHitCount()).isEqualTo(10);
+    }
+
+    private SessionFactory createSessionFactory(Properties props) {
+        Configuration conf = new Configuration();
+        conf.configure(HibernateIT.class.getClassLoader().getResource("test-hibernate-client.cfg.xml"));
+        conf.addAnnotatedClass(AnnotatedEntity.class);
+        conf.addProperties(props);
+
+        final SessionFactory sessionFactory = conf.buildSessionFactory();
+        sessionFactory.getStatistics().setStatisticsEnabled(true);
+        return sessionFactory;
+    }
+
+    private Properties getCacheProperties() {
+        Properties props = new Properties();
+        props.setProperty(Environment.CACHE_REGION_FACTORY, HazelcastCacheRegionFactory.class.getName());
```

Comment:
Should we also test with the `HazelcastLocalCacheRegionFactory`?

Reply:
Sure, added: https://github.com/hazelcast/hazelcast/pull/22282/commits/f669da981562284a5110efe5f53711a4b54fe620